test/test_metalearning/pyMetaLearn/test_meta_features.py

```diff
@@ -109,28 +109,30 @@ def meta_train_data_transformed(request):
     logger = logging.getLogger("Meta")
     meta_features.helper_functions.set_value(
         "MissingValues",
-        meta_features.helper_functions["MissingValues"](X, y, logger, categorical),
+        meta_features.helper_functions["MissingValues"](X, y, logger, feat_type),
     )
     meta_features.helper_functions.set_value(
         "NumSymbols",
-        meta_features.helper_functions["NumSymbols"](X, y, logger, categorical),
+        meta_features.helper_functions["NumSymbols"](X, y, logger, feat_type),
     )
     meta_features.helper_functions.set_value(
         "ClassOccurences",
         meta_features.helper_functions["ClassOccurences"](X, y, logger),
     )
 
-    DPP = FeatTypeSplit(
-        feat_type={
-            col: "categorical" if category else "numerical"
-            for col, category in categorical.items()
-        }
-    )
+    DPP = FeatTypeSplit(feat_type=feat_type)
     X_transformed = DPP.fit_transform(X)
 
-    number_numerical = np.sum(~np.array(list(categorical.values())))
+    number_numerical = np.sum(
+        [True if feat_type[i] == "numerical" else False for i in feat_type.keys()]
+    )
+    number_string = np.sum(
+        [True if feat_type[i] == "string" else False for i in feat_type.keys()]
+    )
     categorical_transformed = {
-        i: True if i < (X_transformed.shape[1] - number_numerical) else False
+        i: True
+        if i < (X_transformed.shape[1] - number_numerical - number_string)
+        else False
```

Comment:
I'm not hugely familiar with this part of the code but it seems the old version relied on categoricals always being the first set of columns? 

```
| f1 | f2 | f3 | f4 | f5 |
+++++++++++++++  ---------
  categorical    numerical
```

The new version seems to follow this and assert an order
```
| f1 | f2 | f3 | f4 | f5 | f6 | f7 | f8
+++++++++++++++ -------------------------
  categorical       string or numerical
```

I really don't know why this is being done in the first place so I can't really say much but it seems like an odd thing to assume?

@mfeurer would know more

Reply:
I don't remember anything like that. What makes you assume that this is the case?

Reply:
```python
X_transformed.shape[1] = 10  # total columns
number_numerical = 6  # number of numerical features

categorical_transformed = {
    i: True if i < (X_transformed.shape[1] - number_numerical) else False
    for i in range(X_transformed.shape[1])
}
# {0: True, 1: True, 2: True, 3: True, 4: False, ..., 9: False }
```

It seems like this previously used to assume some ordering of the features to say the first 4 columns (0-3) were `catergorical_transformed`. I might just be misunderstanding what this really does though.

Reply:
i also was curios about some parts in the code in `metafeatures.py:1145`

```python
X_transformed = DPP.fit_transform(X)
categorical_transformed = {
    i: False for i in range(X_transformed.shape[1])
}
```
so here all columns get transformed to False.

Reply:
So we some should add a test which also contains strings to check this case for all three data types